### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by luisdavim**
To help with the review I've created separate commits for:
* Fixing the hook annotations in the helm release
* Passing down the `debug` flag to helm
* Running the code through `gofumpt`
* Adding retries when using `curl`; sometimes the tests were failing because of that
* Improving error logging for file validation; hopefully it will help track down #567 

ps: We need a better way of asserting error types than comparing strings in our tests....

### CODE REVIEW

File: internal/app/kube_helpers.go [hunk 1]
```diff
@@ -121,7 +120,7 @@ spec:
 
 	definition = definition + Indent(string(d), strings.Repeat(" ", 4))
 	targetFile := path.Join(createTempDir(tempFilesDir, "tmp"), "temp-LimitRange.yaml")
-	if err := ioutil.WriteFile(targetFile, []byte(definition), 0666); err != nil {
+	if err := ioutil.WriteFile(targetFile, []byte(definition), 0o666); err != nil {
```
Comment (jmontroy90) [inline internal/app/kube_helpers.go:123]: `0o666`?

File: internal/app/kube_helpers.go [hunk 2]
```diff
@@ -165,7 +163,7 @@ spec:
 
 	definition = definition + Indent(string(d), strings.Repeat(" ", 4))
 
-	if err := ioutil.WriteFile("temp-ResourceQuota.yaml", []byte(definition), 0666); err != nil {
+	if err := ioutil.WriteFile("temp-ResourceQuota.yaml", []byte(definition), 0o666); err != nil {
```
Comment (jmontroy90) [inline internal/app/kube_helpers.go:166]: `0o666`?

File: internal/app/state_test.go [hunk 3]
```diff
@@ -8,7 +8,7 @@ import (
 
 func setupTestCase(t *testing.T) func(t *testing.T) {
 	t.Log("setup test case")
-	os.MkdirAll(tempFilesDir, 0755)
+	os.MkdirAll(tempFilesDir, 0o755)
```
Comment (jmontroy90) [inline internal/app/state_test.go:11]: `0o755`

File: internal/app/utils.go [hunk 4]
```diff
@@ -60,7 +60,7 @@ func substituteVarsInYaml(file string) string {
 
 	// output file contents with env variables substituted into temp files
 	outFile := path.Join(dir, filepath.Base(file))
-	err = ioutil.WriteFile(outFile, []byte(yamlFile), 0644)
+	err = ioutil.WriteFile(outFile, []byte(yamlFile), 0o644)
```
Comment (jmontroy90) [inline internal/app/utils.go:63]: `0o644`

File: internal/app/utils.go [hunk 5]
```diff
@@ -262,7 +262,7 @@ func copyFile(source string, destination string) {
 	}
 	defer from.Close()
 
-	to, err := os.OpenFile(destination, os.O_RDWR|os.O_CREATE, 0666)
+	to, err := os.OpenFile(destination, os.O_RDWR|os.O_CREATE, 0o666)
```
Comment (jmontroy90) [inline internal/app/utils.go:265]: `0o666`

File: internal/app/utils.go [hunk 6]
```diff
@@ -368,7 +367,7 @@ func replaceStringInFile(input []byte, outfile string, replacements map[string]s
 		output = bytes.Replace(output, []byte(k), []byte(v), -1)
 	}
 
-	if err := ioutil.WriteFile(outfile, output, 0666); err != nil {
+	if err := ioutil.WriteFile(outfile, output, 0o666); err != nil {
```
Comment (jmontroy90) [inline internal/app/utils.go:370]: `0o666`

File: internal/gcs/gcs.go [hunk 7]
```diff
@@ -27,8 +27,7 @@ func Auth() (string, error) {
 		credFile := "/tmp/gcloud_credentials.json"
 		// write the credentials content into a json file
 		d := []byte(os.Getenv("GCLOUD_CREDENTIALS"))
-		err := ioutil.WriteFile(credFile, d, 0644)
-
+		err := ioutil.WriteFile(credFile, d, 0o644)
```
Comment (jmontroy90) [inline internal/gcs/gcs.go:30]: `0o644`

**Review by jmontroy90 — CHANGES_REQUESTED** [review]
Unless you're doing something fancy, all these `0o644` things should be fixed.

**Review by jmontroy90 — APPROVED** [review]

**Comment by luisdavim** [comment]
> Unless you're doing something fancy, all these `0o644` things should be fixed.
> 
> EDIT: octal, coool

yeah I used `gofumpt ` to lint the code and it made that change because:
```
Octal integer literals should use the 0o prefix on modules using Go 1.13 and later
```

**Pull request merged**